### PR TITLE
feat(indexer): Add cli app

### PR DESCRIPTION
**Pull request by roninjin10**
- Add a new cli app
- indexer api runs api
- indexer indexer runs indexer
  - open to a better name
- Both load config toml file
- hook up to actual services in followup prs

### CODE REVIEW

**Comment by roninjin10** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5828** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5828" target="_blank">Graphite</a> 
    * **PR #5829** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5829" target="_blank">Graphite</a> 
      * **PR #5833** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5833" target="_blank">Graphite</a> 
        * **PR #5917** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5917" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5917?utm_source=stack-comment).

File: indexer/cli/cli.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+		),
+	)
+
+	flags := []cli.Flag{
+		ConfigFlag,
+	}
+
+	app := &cli.App{
+		Version:     fmt.Sprintf("%s-%s", GitVersion, params.VersionWithCommit(GitCommit, GitDate)),
+		Description: "Service for indexing deposits and withdrawals by account",
```
Comment (hamdiallam) [inline indexer/cli/cli.go:87]: lets make the description a bit more generic. Maybe something like
"An indexer of all optimism events with a serving api layer"

Comment (roninjin10) [reply indexer/cli/cli.go:87]:
```suggestion
		Description: "An indexer of all optimism events with a serving api layer",
```

File: indexer/cmd/indexer-refresh/main.go [hunk 2]
```diff
@@ -20,34 +14,8 @@ var (
...
-	log.Debug("Loaded config", "config", conf)
-
-	app := cli.NewApp()
-	app.Flags = flags.Flags
-	app.Version = fmt.Sprintf("%s-%s", GitVersion, params.VersionWithCommit(GitCommit, GitDate))
-	app.Name = "indexer"
-	app.Usage = "Indexer Service"
-	app.Description = "Service for indexing deposits and withdrawals " +
-		"by account on L1 and L2"
+	app := cli.NewCli(GitVersion, GitCommit, GitDate)
```
Comment (roninjin10) [inline indexer/cmd/indexer-refresh/main.go:17]:
```suggestion
	app := cli.NewCli(GitVersion, GitCommit, GitDate)
	app.Run()
```
Oops forgot to call Run()

Comment (roninjin10) [reply indexer/cmd/indexer-refresh/main.go:17]: nvm I called it directly below this comment

**Pull request merged**